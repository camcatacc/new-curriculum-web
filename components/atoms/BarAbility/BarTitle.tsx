// Modules
import React from "react";

// Elements
import { Typography } from "@material-ui/core";

// Definitions
export interface BarTitleProps {
	backgroundColor: string;
	fontColor?: string;
	title: any;
	fontSize?: number | string;
}

// Element
const BarTitle = ({ backgroundColor, title, fontSize = "1rem", fontColor = "white" }: BarTitleProps) => {
	return (
		<div className="pl-4 pr-4 pt-2 pb-2" style={{ backgroundColor: backgroundColor }}>
			<Typography variant="body1" style={{ color: fontColor, fontSize: fontSize }} className="w-full truncate" align="center">
				{title}
			</Typography>
		</div>
	);
};

export default BarTitle;
