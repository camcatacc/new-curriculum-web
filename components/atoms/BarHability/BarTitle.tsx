import { Typography } from "@material-ui/core";
import React from "react";

export const BarTitle = (props: {
    backgroundColor: string;
    fontColor: string;
    children: any;
    fontSize?: number | string;
}) => {
    const { backgroundColor, fontColor, fontSize, children } = props;
    return (
        <div
            className="pl-4 pr-4 pt-2 pb-2"
            style={{ backgroundColor: backgroundColor }}
        >
            <Typography
                variant="body1"
                style={{ color: fontColor, fontSize: fontSize }}
                className="w-full truncate"
                align="center"
            >
                {children}
            </Typography>
        </div>
    );
};
