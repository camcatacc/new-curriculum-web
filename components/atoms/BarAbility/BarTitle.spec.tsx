// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import BarTitle from "components/atoms/BarAbility/BarTitle";

// Definitions
import type { BarTitleProps } from "components/atoms/BarAbility/BarTitle";

// Default props
const defaultProps: BarTitleProps = {
	backgroundColor: "red",
	fontColor: "white",
	title: "Testing"
};

// Tests
describe("BarTitle", () => {
	it("Renders as expected", () => {
		const { container } = render(<BarTitle {...defaultProps}>Hello, hello</BarTitle>);
		expect(container).toMatchSnapshot();
	});
});
