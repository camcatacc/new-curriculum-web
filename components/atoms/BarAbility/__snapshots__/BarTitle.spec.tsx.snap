// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarTitle Renders as expected 1`] = `
<div>
  <div
    class="sc-bdvvtL jJpwjr pl-2 pr-2 pt-1 pb-1"
  >
    <p
      class="MuiTypography-root sc-gsDKAQ etnubE w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
      font-size="1rem"
      fontcolor="white"
    >
      Testing
    </p>
  </div>
</div>
`;
